### PR TITLE
Add Auto Moderation events payloads

### PR DESCRIPTION
**Pull request by Victorsitou**
Adds `AUTO_MODERATION_ACTION_EXECUTION` event payload and `AUTO_MODERATION_RULE_CREATE/UPDATE/DELETE` for all auto moderation trigger types. Special thanks to shiftinv for helping me with some payloads :)

reference: https://github.com/discord/discord-api-docs/pull/4860

todo:
- [x] Update `AUTO_MODERATION_ACTION_EXECUTION`

### CODE REVIEW

File: events/guild/auto_mod/auto_moderation_action_execution.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "t": "AUTO_MODERATION_ACTION_EXECUTION",
+    "s": 1,
+    "op": 0,
+    "d": {
+        "type": 2,
+        "metadata": {
+            "channel_id": "100000000000000000"
+        }
```
Comment (shiftinv) [inline events/guild/auto_mod/auto_moderation_action_execution.json]:
```suggestion
        "guild_id": "100000000000000000",
        "action": {
            "type": 2,
            "metadata": {
                "channel_id": "200000000000000000"
            }
        }
```

**Review by shiftinv — COMMENTED** [review]
nice :o
The automod rule objects also have `id`, `guild_id` and `creator_id` fields (I know this is still wip though, hope you don't mind)
I can send you a dump of the complete payloads, but I believe those three are the only missing keys atm c:

File: events/guild/auto_moderation/auto_moderation_action_execution.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            "type": 2,
+            "metadata": {
+                "channel_id": "200000000000000000"
+            }
+        },
+        "rule_id": "300000000000000000",
+        "rule_trigger_type": 1,
+        "channel_id": "40000000000000000",
+        "message_id": null,
+        "alert_system_message_id": "500000000000000000",
```
Comment (Victorsitou) [inline events/guild/auto_moderation/auto_moderation_action_execution.json]:
```suggestion
        "message_id": "500000000000000000",
        "alert_system_message_id": "600000000000000000",
```

File: events/guild/auto_moderation/keyword_preset_auto_moderation_rule_create.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        "creator_id": "300000000000000000",
+        "event_type": 1,
+        "trigger_type": 4,
+        "trigger_metadata": {
+            "keyword_lists": [
+                "SLURS",
+                "SEXUAL_CONTENT",
+                "PROFANITY"
+            ]
+        },
```
Comment (Victorsitou) [inline events/guild/auto_moderation/keyword_preset_auto_moderation_rule_create.json]:
```suggestion
        "trigger_metadata": {
            "presets": [
                1,
                2,
                3
            ]
        },
```

File: events/guild/auto_moderation/keyword_preset_auto_moderation_rule_delete.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        "creator_id": "300000000000000000",
+        "event_type": 1,
+        "trigger_type": 4,
+        "trigger_metadata": {
+            "keyword_lists": [
+                "SLURS",
+                "SEXUAL_CONTENT",
+                "PROFANITY"
+            ]
+        },
```
Comment (Victorsitou) [inline events/guild/auto_moderation/keyword_preset_auto_moderation_rule_delete.json]:
```suggestion
        "trigger_metadata": {
            "presets": [
                1,
                2,
                3
            ]
        },
```

File: events/guild/auto_moderation/keyword_preset_auto_moderation_rule_update.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        "creator_id": "300000000000000000",
+        "event_type": 1,
+        "trigger_type": 4,
+        "trigger_metadata": {
+            "keyword_lists": [
+                "SLURS",
+                "SEXUAL_CONTENT",
+                "PROFANITY"
+            ]
+        },
```
Comment (Victorsitou) [inline events/guild/auto_moderation/keyword_preset_auto_moderation_rule_update.json]:
```suggestion
        "trigger_metadata": {
            "presets": [
                1,
                2,
                3
            ]
        },
```

**Pull request merged**